ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -109,12 +149,212 @@ private BlobSidecarsAndValidationResult internalValidate(final List<BlobSidecar>
     return BlobSidecarsAndValidationResult.validResult(blobSidecars);
   }
 
+  /**
+   * Step 1 of 2
+   *
+   * <p>This function performs a partial validation with blobs that are currently available from the
+   * tracker. If all blobs are already available, it performs a full validation.
+   *
+   * @return a validation result only in case it is a definitive result. If the validation needs to
+   *     be completed it returns empty
+   */
+  private Optional<BlobSidecarsAndValidationResult> validatePartially() {
+    final List<KZGCommitment> kzgCommitmentsInBlock = kzgCommitmentsSupplier.get();
+
+    final List<KZGCommitment> kzgCommitmentsToValidate;
+    final List<BlobSidecar> blobSidecarsToValidate;
+
+    final boolean performCompleteValidation = blockBlobSidecarsTracker.isCompleted();
```

Comment:
I think it isn't a problem because the tracker becomes completed only when is guaranteed to expose the full set of blobSidecar. The tracker is "add only" so it isn't possible that in the meantime a blob disappears.